drivers/spi/spi_ll_stm32.c

```diff
@@ -56,24 +57,48 @@ static inline u8_t spi_stm32_next_tx(struct spi_stm32_data *data)
 /* Shift a SPI frame as master. */
 static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
 {
-	u8_t tx_frame;
-	u8_t rx_frame;
+	u16_t tx_frame;
+	u16_t rx_frame;
+
+	if( spi_context_tx_on(&data->ctx) ){
+		if (SPI_WORD_SIZE_GET(data->ctx.config->operation) == 8) {
+			tx_frame = UNALIGNED_GET((u8_t *)(data->ctx.tx_buf));
+		} else {
+			tx_frame = UNALIGNED_GET((u16_t *)(data->ctx.tx_buf));
+		}
+	} else {
+		tx_frame = SPI_STM32_TX_NOP;
+	}
 
-	tx_frame = spi_stm32_next_tx(data);
```

Comment:
if you remove spi_stm32_next_tx() this looks like the only place it was used, so remove spi_stm32_next_tx function as well.

Reply:
I think it's also used in spi_stm32_shift_s.
Then the question: should the change be extended to slave behavior? And, if yes how to test?
@mbolivar : Maybe you have an idea how to test slave behavior?

Reply:
This function is also called in slave mode (from spi_stm32_shift_s).
For now, my patch is only for master mode.

Reply:
I think that rather than removing spi_stm32_next_tx(), the above lines should be moved into spi_stm32_next_tx(), which should be modified to return a u16_t. The purpose of this routine is to return the next TX frame.

@erwango I unfortunately don't have an immediate answer for how to test slave mode as we only use master mode here.

Looking at the existing tests, spi_loopback is master mode only, and spi_basic_api relies on the SPI_MODE_LOOP flag, which I don't believe the STM32 IP can support (please correct me if I'm wrong). So fixing SPI slave testing in Zephyr would appear to be a more general problem.

However, IMO the code in spi_stm32_shift_s() is simple enough that it should be updated now as  well.